frontend/src/pages/CodeEditor.tsx

```diff
@@ -65,40 +80,61 @@ const CodeEditor = () => {
     if (sessionId) {
       getSessionAndWsTicket(sessionId)
         .then((resp) => {
-          const { question, ticket } = resp.data;
+          const { question } = resp.data;
           setQuestion(question);
-          setWsTicket(ticket);
         })
         .catch(() => {
           throwAsyncError('Invalid session');
         });
+
+      getSessionLanguage(sessionId).then((resp) => {
+        setSelectedLanguage(resp.data);
+      });
     }
+
+    return () => {
+      setEditor(null);
+    };
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
   // Initialize websocket connection to Yjs service
   useEffect(() => {
+    if (selectedLanguage === placeholderLanguage) return;
+
     let provider: WebsocketProvider | null = null;
-    if (editor && wsTicket) {
-      // Handles websocket disconnection when unauthorized (e.g. invalid ws ticket)
-      provider = bindYjsToMonacoEditor(wsTicket, editor, throwAsyncError);
+
+    if (editor && sessionId) {
+      getSessionAndWsTicket(sessionId)
+        .then((resp) => {
+          const { ticket } = resp.data;
+          provider = bindYjsToMonacoEditor(ticket, selectedLanguage.id, editor, throwAsyncError);
```

Comment:
Is there a reason to separate these useEffects? Also we should take the language from the backend instead of the frontend